ls1intum/Artemis
# [Feature] Show Student Login on Submission and Assessment Page

### Checklist
#### General



#### Client

### Motivation and Context


This PR fixes the Issue #3974 

### Description

The student login is shown in a separate column next to the students name on the Submission and Assessment Page.

### Steps for Testing

Prerequisites:
- 1 Instructor
- at least two Student
- 1 Programming, Text, Modeling, File Upload Exercise and an Exam with at least one Exercise per type with at least two student submissions

1. Log in to Artemis as Instructor
2. Navigate to Course Administration
3. Create the above mentioned exercises and the exam or use existing ones which fulfill the prerequisites
4. Login as Student
5. Submit something with at least two students in all the regular and exam exercises. If you already have such submissions you can skip this step.
6. Log in with the Instructor again
7. Do the following steps for all the exercises:
    7.1 Navigate to the Submissions and Assessments Page by clicking the "Submissions"-Button
    7.2 Check the table of the submissions you did in step 5. The table should show the students login next to the name of the student
    7.3 Click the sort item in the table headline row next to "Login" to sort the entries based on the student login 

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots



This screenshot shows the Submissions and Assessments Page before the changes:
<img width="1440" alt="Before" src="https://user-images.githubusercontent.com/82283581/135622584-763462c0-86b0-41d2-85ad-aab7f3d718a8.png">

This screenshot shows the Submissions and Assessments Page after the changes: The students login is displayed in a separate column next to the name:
<img width="1440" alt="After" src="https://user-images.githubusercontent.com/82283581/135622716-9f81d180-8b8e-4e6a-b9a8-8ef8150d6d05.png">

This short video shows the sorting based on the student login:
https://user-images.githubusercontent.com/82283581/135622796-63e187cf-055a-4db9-9058-6b8129c7f7a9.mov
